test/token/ERC20/extensions/ERC20Votes.test.js

```diff
@@ -590,4 +592,44 @@ contract('ERC20Votes', function (accounts) {
       });
     });
   }
+
+  describe('with quadratic voting units', function () {
+    beforeEach(async function () {
+      this.token = await ERC20VotesQuadratic.new(name, symbol, name, version);
+      this.votes = this.token;
+    });
+
+    it('correctly transfers voting units', async function () {
+      const alice = holder;
+      const bob = holderDelegatee;
+      const charlie = other1;
+
+      const tokens = 100;
+      const votingUnits = 10_000;
+
+      // Alice delegates 100 votes to bob
+      await this.token.delegate(bob, { from: alice });
+      await this.token.$_mint(alice, tokens);
+
+      // As a result, bob has (tokens ** 2) voting units
+      expect(await this.token.getVotes(bob)).to.be.bignumber.eq(votingUnits.toString());
+
+      // Now alice transfers her tokens to charlie who delegates to himself
+      await this.token.delegate(charlie, { from: charlie });
+      const { receipt } = await this.token.transfer(charlie, tokens, { from: alice });
+
+      // As a result, charlie got transferred (tokens ** 2) voting units
+      expectEvent(receipt, 'Transfer', { from: alice, to: charlie, value: tokens.toString() });
+      expectEvent(receipt, 'DelegateVotesChanged', {
+        delegate: bob,
+        previousUnits: votingUnits.toString(),
+        newUnits: '0',
+      });
+      expectEvent(receipt, 'DelegateVotesChanged', {
+        delegate: charlie,
+        previousUnits: '0',
+        newUnits: votingUnits.toString(),
+      });
+    });
+  });
```

Comment:
This will not work if the transferred amount is not the entire user balance. IMO we should remove it, and add warning so that user don't try to do exactly that.

Reply:
What are you proposing to remove? The tests, the ERC20VotesQuadratic mock, or the _toVotingUnits function?

Reply:
At the very least the quadratic mock and tests.

Reply:
Since the only think we support are linear functions. We could have a "rate" that would be 1:1 by default and that we could increase (for example 5 voting unit per token). 

If we do that we would have to be carefull of rounding errors. Rate should probably be an integer. It would have limited value for ERC20 but maybe it would be nice for some erc721 ?

Reply:
>we would have to be carefull of rounding errors

Yeah this is another thing implementors of `_toVotingUnits` functions would have to keep in mind... Quite error prone. I don't think we should add the rate mechanism either, it's unrelated to the comments in the audit.

For ERC-721 we've talked about this but we were concerned that if the rate is a function people might implement it in a way that isn't constant and would cause errors. I guess if it's a pure function with no arguments it would need to be constant, but not if the contract is upgradeable.

Reply:
>It might be better to remove _toVotingUnits altogether and improve the documentation to make sure users use caution.

👍🏻

Reply:
I didn't realize this doesn't fully work because of partial transfers but the example from @Amxx was clear enough. I agree with removing the whole tests for quadratic voting and I'll add a note in that regard.

Reply:
Note added and changes rolled back